core/src/main/scala/org/apache/spark/status/protobuf/DeterministicLevelSerializer.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.status.protobuf
+
+import org.apache.spark.rdd.DeterministicLevel
+import org.apache.spark.status.protobuf.StoreTypes.{DeterministicLevel => GDeterministicLevel}
+
+private[protobuf] object DeterministicLevelSerializer {
+
+  def serialize(input: DeterministicLevel.Value): GDeterministicLevel = {
```

Comment:
I still want to know which of the following code and case match is faster

```scala
  private lazy val scalaToPb = Map(
    DeterministicLevel.DETERMINATE -> GDeterministicLevel.DETERMINISTIC_LEVEL_DETERMINATE,
    DeterministicLevel.UNORDERED -> GDeterministicLevel.DETERMINISTIC_LEVEL_UNORDERED,
    DeterministicLevel.INDETERMINATE -> GDeterministicLevel.DETERMINISTIC_LEVEL_INDETERMINATE
  )

  def serialize(input: DeterministicLevel.Value): GDeterministicLevel = scalaToPb(input)
```

Reply:
A simple test is as follows:

https://github.com/LuciferYang/spark/commit/3d4dc25eb03ba35ad2497f2d1a2274300e1cc7ee

run with GA: https://github.com/LuciferYang/spark/actions/runs/3816561346

`case match` is slow:

```
OpenJDK 64-Bit Server VM 1.8.0_352-b08 on Linux 5.15.0-1023-azure
Intel(R) Xeon(R) Platinum 8272CL CPU @ 2.60GHz
Test serialize:                           Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
------------------------------------------------------------------------------------------------------------------------
Use map                                             192            195           3          5.2         192.4       1.0X
Use case match                                      454            455           1          2.2         453.8       0.4X
```

Reply:
Could you run it again in reverse order？
Because when I run as follow, result:
<img width="1050" alt="image" src="https://user-images.githubusercontent.com/15246973/210170542-b3c842d1-1527-483f-825f-0151a678c2cc.png">
It is very weird!

Reply:
When I run it on GA, result as follow:
<img width="955" alt="image" src="https://user-images.githubusercontent.com/15246973/210172158-d4acf249-e08b-47ca-8f81-9fa9f579e0c0.png">
https://github.com/panbingkun/spark/actions/runs/3816951445/jobs/6492899136
Code:
https://github.com/apache/spark/pull/39286/files#diff-eca16ee278b786f4ab866072396455b93c6bedd8e3b8d562935623931ae3936cR36-R58

Reply:
OK, I see , it fine to me to use `case match` now.  I think we should investigate why the result data changes with the test order , but this does not block this pr